github/commitstatus.go

```diff
@@ -262,35 +259,46 @@ func min(a, b int) int {
 //  2. The HTTP status code is in a retryable subset of the 5xx status codes.
 //     In this case, it returns the same as the input parameter waitTime.
 func checkForRetry(res httpResponse, waitTime, maxSleepTime, jitter time.Duration,
-) (time.Duration, string, error) {
+) (time.Duration, string) {
 	retryableStatusCodes := []int{
 		http.StatusInternalServerError, // 500
 		http.StatusBadGateway,          // 502
 		http.StatusServiceUnavailable,  // 503
 		http.StatusGatewayTimeout,      // 504
 	}
 
-	switch {
+	// Are we rate limited ?
 	// If the request exceeds the rate limit, the response will have status 403 Forbidden
 	// and the x-ratelimit-remaining header will be 0
 	// https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#exceeding-the-rate-limit
-	case res.statusCode == http.StatusForbidden && res.rateLimitRemaining == 0:
+	if res.statusCode == http.StatusForbidden && res.rateLimitRemaining == 0 {
 		// Calculate the sleep time based solely on the server clock. This is unaffected
 		// by the inevitable clock drift between server and client.
 		sleepTime := res.rateLimitReset.Sub(res.date)
+		// Be robust to possible races in the GitHub backend. This avoids failing too early.
+		if sleepTime < 0 {
+			sleepTime = 0
+		}
 		// Be a good netizen by adding some jitter to the time we sleep.
 		sleepTime += jitter
+
 		switch {
+		case sleepTime == 0:
+			return 0, ""
```

Comment:
I would add a msg here because this will terminate the loop without succesfull request, so the situation is quite different than:
```
case sleepTime > maxSleepTime:
   return 0, ""
```

Reply:
There is no message because the current client code would not print such message.

> because this will terminate the loop without successful request

Ah. Let me check.

Reply:
Ok. I see why there is no msg.

But as we discussed in person, if the request failed, we would break the loop and the next attempt would not happen.
and in `checkStatus` we would inform the user we failed due to being rate limited and you had to wait just 0 seconds.

Reply:
Resolving conversation as the code changed a lot, and the changes address also this.